app/src/main/java/com/snatik/matches/fragments/DifficultySelectFragment.java

```diff
@@ -50,9 +63,120 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
 
 		animate(difficulty1, difficulty2, difficulty3, difficulty4, difficulty5, difficulty6);
 
+		Typeface type = Typeface.createFromAsset(Shared.context.getAssets(),"fonts/grobold.ttf");
+
+
+        TextView text1 = (TextView) view.findViewById(R.id.time_difficulty_1);
+		text1.setGravity(Gravity.CENTER_HORIZONTAL | Gravity.CENTER_VERTICAL);
+		text1.setTypeface(type);
+
+        TextView text2 = (TextView) view.findViewById(R.id.time_difficulty_2);
+		text2.setGravity(Gravity.CENTER_HORIZONTAL | Gravity.CENTER_VERTICAL);
+		text2.setTypeface(type);
+
+        TextView text3 = (TextView) view.findViewById(R.id.time_difficulty_3);
+		text3.setGravity(Gravity.CENTER_HORIZONTAL | Gravity.CENTER_VERTICAL);
+		text3.setTypeface(type);
+
+        TextView text4 = (TextView) view.findViewById(R.id.time_difficulty_4);
+		text4.setGravity(Gravity.CENTER_HORIZONTAL | Gravity.CENTER_VERTICAL);
+		text4.setTypeface(type);
+
+        TextView text5 = (TextView) view.findViewById(R.id.time_difficulty_5);
+		text5.setGravity(Gravity.CENTER_HORIZONTAL | Gravity.CENTER_VERTICAL);
+		text5.setTypeface(type);
+
+        TextView text6 = (TextView) view.findViewById(R.id.time_difficulty_6);
+		text6.setGravity(Gravity.CENTER_HORIZONTAL | Gravity.CENTER_VERTICAL);
+		text6.setTypeface(type);
+
+
+
+		SQLiteDB db = new SQLiteDB(Shared.context,null,null,1);
+        List <Integer> times = new ArrayList<>();
+
+		for(int i = 1; i<=6 ; i++) {
+			if(db.getBestTimeForStage(theme.id, i)!= 0) {
+				times.add(db.getBestTimeForStage(theme.id, i));
+			} else {
+				times.add(-1);
+			}
+			System.out.println(times.get(i-1));
+		}
+
+		if(times.get(0) != -1) {
+
+			SimpleDateFormat dateFormat = new SimpleDateFormat("mm:ss");
+			dateFormat.setTimeZone(TimeZone.getTimeZone("GMT"));
+			Date date = new Date(times.get(0) * 1000);
+			text1.setText(" BEST : " + dateFormat.format(date));
+
+		} else {
+			text1.setText("BEST :  - ");
+		}
+
+		if(times.get(1) != -1) {
```

Comment:
It's a lot of duplicated code and logic for all `textView1..6`. 

Instead of all these lines of code. Go back to place where you initialize and set text views and set the best time there. For example for text2:

```
TextView text2 = (TextView) view.findViewById(R.id.time_difficulty_2);
text2.setGravity(Gravity.CENTER_HORIZONTAL | Gravity.CENTER_VERTICAL);
text2.setTypeface(type);
text2.setText(getBestTime(2, ..));
```

And now create new function called: `getBestTime(difficulty, theme)` that returns String. The function will read the time from `Memory` and create the string "Best : - " or "Best : 01:49".